modules/src/ics04_channel/handler/chan_open_try.rs

```diff
@@ -0,0 +1,479 @@
+               && old_channel_end.version_matches(&msg.channel.version())
+            {
+                // A ChannelEnd already exists and all validation passed.
+                Ok(old_channel_end)
+            } else {
+                // A ConnectionEnd already exists and validation failed.
+                Err(Into::<Error>::into(
+                    Kind::ChannelMismatch(prev_id.clone()).context(
+                        old_channel_end
+                            .counterparty()
+                            .channel_id()
+                            .clone()
+                            .unwrap()
+                            .to_string(),
+                    ),
+                ))
+            }
+        }
+        // No channel id is supplied to  create a new channel end. Note: the id is assigned
+        // by the ChannelKeeper.
+        None => {
+            channel_id = None;
+
+            Ok(ChannelEnd::new(
+                State::Init,
+                *msg.channel.ordering(),
+                msg.channel.counterparty().clone(),
+                msg.channel.connection_hops().clone(),
+                msg.counterparty_version().clone(),
+            ))
+        }
+    }?;
+
+    // An IBC connection running on the local (host) chain should exist.
+
+    if msg.channel.connection_hops().len() != 1 {
+        return Err(Kind::InvalidConnectionHopsLength.into());
+    }
+
+    let connection_end = ctx.connection_state(&msg.channel().connection_hops()[0]);
+
+    let conn = connection_end
+        .ok_or_else(|| Kind::MissingConnection(msg.channel().connection_hops()[0].clone()))?;
+
+    if !conn.state_matches(&ConnectionState::Open) {
+        return Err(ConnectionNotOpen(msg.channel.connection_hops()[0].clone()).into());
+    }
+
+    let get_versions = conn.versions();
+    let version = match get_versions.as_slice() {
+        [version] => version,
+        _ => return Err(Kind::InvalidVersionLengthConnection.into()),
+    };
+
+    let channel_feature = msg.channel().ordering().as_string().to_string();
+    if !version.is_supported_feature(channel_feature) {
+        return Err(Kind::ChannelFeatureNotSuportedByConnection.into());
+    }
+
+    //Channel capabilities
+    let cap = ctx.port_capability(&msg.port_id().clone());
+    let cap_key;
+
+    match cap {
+        Some(key) => {
+            if !ctx.capability_authentification(&msg.port_id().clone(), &key) {
+                return Err(Kind::InvalidPortCapability.into());
+            } else {
+                cap_key = key;
+            }
+        }
+        None => return Err(Kind::NoPortCapability.into()),
+    }
+
+    if msg.channel().version().is_empty() {
+        return Err(Kind::InvalidVersion.into());
+    }
+
+    // Proof verification in two steps:
+    // 1. Setup: build the Channel as we expect to find it on the other party.
+
+    let _expected_counterparty = Counterparty::new(msg.port_id().clone(), None);
```

Comment:
```suggestion
    let expected_counterparty = Counterparty::new(msg.port_id().clone(), None);
```
let's use this below